tests/util/helpers.sh

```diff
@@ -108,19 +108,35 @@ _wait_for_statefulset() {
 # Wait for Istio config to propagate
 # usage: _wait_for_istio <kind> <namespace> <name>
 _wait_for_istio() {
-    #local kind="$1"
-    #local namespace="$2"
-    #local name="$3"
-    local start=$(date +%s)
-    sleep 1s
-    # @TODO: Rewrite this to *NOT* use istioctl experimental wait, since it was removed
-    # https://github.com/istio/istio.io/issues/16429
-    #if ! istioctl experimental wait --for=distribution --timeout=10s "$kind" "$name.$namespace"; then
-        #echo "Failed distribution of $kind $name in namespace $namespace"
-        #istioctl ps
-        #echo "TEST: wait for failed, but continuing."
-    #fi
-    echo "Duration: $(($(date +%s)-start)) seconds"
+    local kind="$1"
+    local namespace="$2"
+    local name="$3"
+    local max_attempts=30
+    local attempt=1
+    local sleep_time=1
+    local found=0
+    local start
+    start="$(date +%s)"
+
+    while (( attempt <= max_attempts )); do
+        # Check if the resource exists
+        if kubectl get "$kind" "$name" -n "$namespace" >/dev/null 2>&1; then
```

Comment:
Aren't you basically doing

`kubectl wait --for=create -n $namespace $kind/$name --timeout=30s`

with this whole function?

Perhaps that would be cleaner?

Reply:
That's what some of the other functions do.

The thing here is, that the goal of this wasn't so much to ensure that the configuration was created in Kubernetes, which either method checks for, but to _ensure all the sidecars were programmed with it_.  They generally are, but in the instance that the object is created but is not translated to XDS and sent to the other proxy servers, there's no obvious way to tell.

So `kubectl wait` is probably necessary but not sufficient.